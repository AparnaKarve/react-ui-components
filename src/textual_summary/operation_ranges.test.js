import React from 'react';
import renderer from 'react-test-renderer';
import OperationRanges from './operation_ranges';

describe('Operation Ranges', () => {
  it('renders just fine...', () => {
    const rangesData = {
      title: 'Operation Ranges',
      items: [
        {
          label: 'CPU',
          value: [{ label: 'Max', value: 'Not Available' }, { label: 'High', value: '0 MHz' }, { label: 'Average', value: '0 MHz' }, { label: 'Low', value: '0 MHz' }],
          hoverClass: 'no-hover',
        },
        {
          label: 'CPU Usage',
          value: [{ label: 'Max', value: 'Not Available' }, { label: 'High', value: '0.00%' }, { label: 'Average', value: '0.00%' }, { label: 'Low', value: '0.00%' }],
          hoverClass: 'no-hover',
        },
      ],
    };
    const component = renderer.create(<OperationRanges title={rangesData.title} items={rangesData.items} />);
    const table = component.toJSON();
    expect(table).toMatchSnapshot();
  });
});

