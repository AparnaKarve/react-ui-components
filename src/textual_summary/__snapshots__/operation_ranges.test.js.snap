// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Ranges renders just fine... 1`] = `
<table
  className="table table-bordered table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        align="left"
        colSpan="3"
      >
        Operation Ranges
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        rowSpan={5}
      >
        CPU
      </td>
      <th>
         
      </th>
      <th>
         
      </th>
    </tr>
    <tr>
      <td>
        Max
      </td>
      <td>
        Not Available
      </td>
    </tr>
    <tr>
      <td>
        High
      </td>
      <td>
        0 MHz
      </td>
    </tr>
    <tr>
      <td>
        Average
      </td>
      <td>
        0 MHz
      </td>
    </tr>
    <tr>
      <td>
        Low
      </td>
      <td>
        0 MHz
      </td>
    </tr>
    <tr>
      <td
        rowSpan={5}
      >
        CPU Usage
      </td>
      <th>
         
      </th>
      <th>
         
      </th>
    </tr>
    <tr>
      <td>
        Max
      </td>
      <td>
        Not Available
      </td>
    </tr>
    <tr>
      <td>
        High
      </td>
      <td>
        0.00%
      </td>
    </tr>
    <tr>
      <td>
        Average
      </td>
      <td>
        0.00%
      </td>
    </tr>
    <tr>
      <td>
        Low
      </td>
      <td>
        0.00%
      </td>
    </tr>
  </tbody>
</table>
`;
